src/install/hosted_git_info.zig

```diff
@@ -0,0 +1,1742 @@
+
+                    return std.fmt.allocPrint(
+                        alloc,
+                        "git+https://{s}/{s}.git{s}{s}",
+                        .{ self.domain(), project, cmsh_sep, cmsh },
+                    );
+                }
+
+                fn sourcehut(
+                    self: Self,
+                    alloc: std.mem.Allocator,
+                    auth: ?[]const u8,
+                    user: ?[]const u8,
+                    project: []const u8,
+                    committish: ?[]const u8,
+                ) error{OutOfMemory}![]u8 {
+                    requiresUser(user);
+                    _ = auth;
+
+                    const cmsh: []const u8 = if (committish) |c| c else "";
+                    const cmsh_sep = if (cmsh.len > 0) "#" else "";
+
+                    return std.fmt.allocPrint(
+                        alloc,
+                        "https://{s}/{s}/{s}.git{s}{s}",
+                        .{ self.domain(), user.?, project, cmsh_sep, cmsh },
+                    );
+                }
+            };
+
+            /// Mirrors hosts.js's shortcuttemplate
+            const Shortcut = struct {
+                const Type = *const fn (
+                    self: Self,
+                    allocator: std.mem.Allocator,
+                    user: ?[]const u8,
+                    project: []const u8,
+                    committish: ?[]const u8,
+                ) error{OutOfMemory}![]u8;
+
+                fn default(
+                    self: Self,
+                    alloc: std.mem.Allocator,
+                    user: ?[]const u8,
+                    project: []const u8,
+                    committish: ?[]const u8,
+                ) error{OutOfMemory}![]u8 {
+                    requiresUser(user);
+
+                    const cmsh: []const u8 = if (committish) |c| c else "";
+                    const cmsh_sep = if (cmsh.len > 0) "#" else "";
+
+                    return std.fmt.allocPrint(
+                        alloc,
+                        "{s}{s}/{s}{s}{s}",
+                        .{ self.shortcut(), user.?, project, cmsh_sep, cmsh },
+                    );
+                }
+
+                fn gist(
+                    self: Self,
+                    alloc: std.mem.Allocator,
+                    user: ?[]const u8,
+                    project: []const u8,
+                    committish: ?[]const u8,
+                ) error{OutOfMemory}![]u8 {
+                    _ = user;
+
+                    const cmsh: []const u8 = if (committish) |c| c else "";
+                    const cmsh_sep = if (cmsh.len > 0) "#" else "";
+
+                    return std.fmt.allocPrint(
+                        alloc,
+                        "{s}{s}{s}{s}",
+                        .{ self.shortcut(), project, cmsh_sep, cmsh },
+                    );
+                }
+            };
+
+            /// Mirrors hosts.js's extract function
+            const Extract = struct {
```

Comment:
most functions in this namespace are eating errors and need `OOM!Result` and `try`

Reply:
Makes sense. I matched `npa`, but agree with what you're saying

Reply:
^ That's a lie, looks like my other functions return `OutOfMemory` just fine, I don't really know why I did it this way :\